backend/infrahub/message_bus/operations/git/repository.py

```diff
@@ -44,3 +45,31 @@ async def import_objects(message: messages.GitRepositoryImportObjects, service:
         )
         repo.task_report = git_report
         await repo.import_objects_from_files(infrahub_branch_name=message.infrahub_branch_name, commit=message.commit)
+
+
+@flow(name="refresh-git-fetch")
+async def fetch(message: messages.RefreshGitFetch, service: InfrahubServices) -> None:
+    if message.meta and message.meta.initiator_id == WORKER_IDENTITY:
```

Comment:
Could we do this check earlier in `infrahub.message_bus.operations.execute_message`